api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3939,17 +4128,55 @@ public void putObject(String bucketName, String objectName, ServerSideEncryption
    *
    * @see #putObject(String bucketName, String objectName, String fileName)
    */
-  public void putObject(String bucketName, String objectName, InputStream stream, String contentType)
-    throws InvalidBucketNameException, NoSuchAlgorithmException, IOException,
-           InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
-           InternalException,
-           InvalidArgumentException, InsufficientDataException {
+  public void putObject(String bucketName, String objectName, Object data, Long size,
```

Comment:
`Object` argument is too generic and it will lead to confusion, hence this should be private method.  All argument `putObject()` should accept `InputStream` and `putObject()` with filename should use Open file then call this private `putObject()`.

Reply:
done.
@balamurugana  PTAL